src/effects/specialeffectchainslots.cpp

```diff
@@ -0,0 +1,197 @@
+    return QString("[EffectRack1_EffectUnit%1]")
+                   .arg(QString::number(iChainNumber + 1));
+}
+
+QString StandardEffectChainSlot::formatEffectSlotGroup(const int iChainSlotNumber,
+                                                       const int iEffectSlotNumber) {
+    return QString("[EffectRack1_EffectUnit%1_Effect%2]")
+                   .arg(QString::number(iChainSlotNumber + 1))
+                   .arg(QString::number(iEffectSlotNumber + 1));
+}
+
+
+OutputEffectChainSlot::OutputEffectChainSlot(EffectsManager* pEffectsManager)
+        : EffectChainSlot(formatEffectChainSlotGroup("[Master]"),
+                          0, pEffectsManager,
+                          SignalProcessingStage::Postfader, true,
+                          formatEffectChainSlotGroup("[Master]")) {
+    addEffectSlot("[OutputEffectRack_[Master]_Effect1]");
+
+    // Register the master channel
+    const ChannelHandleAndGroup* masterHandleAndGroup = nullptr;
+
+    // TODO(Be): Remove this hideous hack to get the ChannelHandleAndGroup
+    const QSet<ChannelHandleAndGroup>& registeredChannels =
+            m_pEffectsManager->registeredInputChannels();
+    for (const ChannelHandleAndGroup& handle_group : registeredChannels) {
+        if (handle_group.name() == "[MasterOutput]") {
+            masterHandleAndGroup = &handle_group;
+            break;
+        }
+    }
+    DEBUG_ASSERT(masterHandleAndGroup != nullptr);
+
+    registerInputChannel(*masterHandleAndGroup);
+    enableForInputChannel(*masterHandleAndGroup);
+    m_pControlChainMix->set(1.0);
+    sendParameterUpdate();
+}
+
+QString OutputEffectChainSlot::formatEffectChainSlotGroup(const QString& group) {
+    return QString("[OutputEffectRack_%1]").arg(group);
+}
+
+
+PerGroupEffectChainSlot::PerGroupEffectChainSlot(const QString& group,
+                                                 const QString& chainSlotGroup,
+                                                 unsigned int iChainNumber,
+                                                 EffectsManager* pEffectsManager)
+        : EffectChainSlot(chainSlotGroup, iChainNumber, pEffectsManager,
+                          SignalProcessingStage::Prefader, false,
+                          chainSlotGroup) {
+    // Set the chain to be fully wet.
+    m_pControlChainMix->set(1.0);
+    sendParameterUpdate();
+
+    // TODO(rryan): remove.
+    const ChannelHandleAndGroup* handleAndGroup = nullptr;
+    for (const ChannelHandleAndGroup& handle_group :
+             m_pEffectsManager->registeredInputChannels()) {
+        if (handle_group.name() == group) {
+            handleAndGroup = &handle_group;
+            break;
+        }
+    }
+    DEBUG_ASSERT(handleAndGroup != nullptr);
+
+    // Register this channel alone with the chain slot.
+    registerInputChannel(*handleAndGroup);
+    enableForInputChannel(*handleAndGroup);
+
+    // Add a single effect slot
+    addEffectSlot(chainSlotGroup);
```

Comment:
This is not working. The Effect group name is not the same as the ChainSlot group name. Use `formatEffectSlotGroup`